sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -2003,6 +2042,92 @@ def set_premium_page_blob_tier(self, premium_page_blob_tier, **kwargs):
         except StorageErrorException as error:
             process_storage_error(error)
 
+    def _set_blob_tags_options(self, tags=None, **kwargs):
+        # type: (Optional[Dict[str, str]], **Any) -> Dict[str, Any]
+        headers = kwargs.pop('headers', {})
+
+        tags = serialize_blob_tags(tags)
+
+        options = {
+            'tags': tags,
+            'version_id': kwargs.pop('version_id', None),
+            'timeout': kwargs.pop('timeout', None),
+            'cls': return_response_headers,
+            'headers': headers}
```

Comment:
I don't think we need to pop timeout, verion_id or headers - these will automatically be passed through with kwargs